src/libawkward/array/UnionArray.cpp

```diff
@@ -322,15 +322,15 @@ namespace awkward {
   UnionArrayOf<T, I>::regular_index(const IndexOf<T>& tags) {
     int64_t lentags = tags.length();
     int64_t size;
-    struct Error err1 = awkward_unionarray8_regular_index_getsize(
+    struct Error err1 = kernel::unionarray_regular_index_getsize<int8_t>(
```

Comment:
This is similar to the `_64` suffix cases: there's only one variant, `awkward_unionarray8_regular_index_getsize`, so its name should not be changed and it should not be templatized.